src/switch_core_media.c

```diff
@@ -3212,15 +3247,20 @@ static void check_media_timeout_params(switch_core_session_t *session, switch_rt
 {
 	switch_media_type_t type = engine->type;
 	const char *val;
+	uint8_t inb = switch_channel_direction(session->channel) == SWITCH_CALL_DIRECTION_INBOUND;
 
 	if ((val = switch_channel_get_variable(session->channel, "media_hold_timeout"))) {
 		engine->media_hold_timeout = atoi(val);
 	}
 
-	if ((val = switch_channel_get_variable(session->channel, "media_timeout"))) {
+	if ((val = switch_channel_get_variable(session->channel, inb ? "media_timeout" : "partner_media_timeout"))) {
 		engine->media_timeout = atoi(val);
 	}
 
+	if ((val = switch_channel_get_variable(session->channel, inb ? "partner_media_timeout" : "media_timeout"))) {
```

Comment:
This is wrong, direction can't be use as to decide if we will user partner or not. What if I originate two channels and bridge them together so both direction would be outbound? What if channel sent a re-invite which changes the direction as well